cpp/csp/core/BasicAllocator.h

```diff
@@ -15,7 +15,7 @@ namespace csp
 {
 
 // Extremely basic non-thread safe fixed-size allocator
-class BasicAllocator
+class CSPCORE_EXPORT BasicAllocator
```

Comment:
seems like most of these classes under core/ are header-only and shouldn't be exported since on windows this means anyone linking against these would link as dllimport; but there's no concrete dllexport for the header-only classes in csp_core.dll